science/nektarpp/Portfile

```diff
@@ -0,0 +1,127 @@
+                       port:boost \
+                       port:fftw-3 \
+                       port:opencascade \
+                       port:scotch \
+                       port:tinyxml \
+                       port:zlib
+
+# Don't build demos and tests by default.
+configure.args-append \
+    -DNEKTAR_BUILD_DEMOS=OFF           \
+    -DNEKTAR_BUILD_TESTS=OFF           \
+    -DNEKTAR_BUILD_UNIT_TESTS=OFF      \
+    -DNEKTAR_SOLVER_DIFFUSION=OFF      \
+    -DNEKTAR_SOLVER_DUMMY=OFF          \
+    -DNEKTAR_SOLVER_DIFFUSION=OFF      \
+    -DNEKTAR_SOLVER_ELASTICITY=OFF     \
+    -DNEKTAR_SOLVER_DIFFUSION=OFF      \
+    -DNEKTAR_SOLVER_MMF=OFF            \
+    -DNEKTAR_USE_ARPACK=ON             \
+    -DNEKTAR_USE_FFTW=ON               \
+    -DNEKTAR_USE_SCOTCH=ON             \
+    -DNEKTAR_USE_MESHGEN=ON
+
+variant demos description {
+    Build additional Nektar++ demos
+} {
+    configure.args-append -DNEKTAR_BUILD_DEMOS=ON
+}
+
+variant petsc description {
+    Enable PETSc support for linear algebra solvers.
+} {
+    configure.args-append -DNEKTAR_USE_PETSC=ON
+    depends_lib-append port:petsc
+}
+
+if {[ mpi_variant_isset ]} {
+    depends_lib-append port:hdf5
+    require_active_variants hdf5 [ mpi_variant_name ]
+    require_active_variants scotch [ mpi_variant_name ]
+    configure.args-append -DNEKTAR_USE_MPI=ON -DNEKTAR_USE_HDF5=ON
+}
+
+variant oce description {
+    use Community Edition version of OpenCASCADE
+} {
+    depends_lib-replace port:opencascade port:oce
+}
+
+set pythons_suffixes    {27 36 37 38}
+
+set pythons_ports       {}
+foreach s ${pythons_suffixes} {
+    lappend pythons_ports python${s}
+}
+
+foreach s ${pythons_suffixes} {
+    set p python${s}
+    set v [string index ${s} 0].[string index ${s} 1]
+    set pe ${prefix}/bin/python${v}
+    set i [lsearch -exact ${pythons_ports} ${p}]
+    set c [lreplace ${pythons_ports} ${i} ${i}]
+    variant ${p} description "Build the Python ${v} bindings" conflicts {*}${c} "
+        depends_lib-append      port:${p} \
+                                port:py${s}-numpy \
+                                port:boost-numpy
+        configure.args-append   -DNEKTAR_BUILD_PYTHON=ON \
+                                -DPYTHON_EXECUTABLE=${pe}
+
+        if {${s} != 27} {
+            configure.args-append -DNEKTAR_USE_PYTHON3=ON -DNEKTAR_PYTHON3_STATUS=ON
+        }
+
+        # Require a separate destroot, since there is a conflict between CMake install prefix
+        # and the requirement for DESTDIR.
+        destroot {
+           set nekpy_libs {\"LibUtilities\" \"StdRegions\" \"SpatialDomains\" \"LocalRegions\" \"MultiRegions\"}
+
+           foreach npl \${nekpy_libs} {
+               system \"cp ${build.dir}/library/\${npl}/Python/_\${npl}.so ${build.dir}/NekPy/\${npl}\"
+           }
+           system \"cd ${build.dir} && DESTDIR=${destroot} make install\"
+           system \"cd ${build.dir} && ${pe} setup.py install --root=${destroot}\"
```

Comment:
Don't use `cd` in the system command. Instead use the `-W` flag to set the directory:

`system -W {dir} {command}
`